gpMgmt/bin/gprebalance

```diff
@@ -0,0 +1,324 @@
+#!/usr/bin/env python3
+
+from gppylib.mainUtils import getProgramName
+
+import psycopg2
+import yaml
+from psycopg2 import DatabaseError, OperationalError
+try:
+    from gppylib.commands.unix import *
+    from gppylib.commands.gp import *
+    from gppylib.gparray import GpArray, Segment
+    from gppylib.gpparseopts import OptParser, OptChecker
+    from gppylib.gplog import *
+    from gppylib.db import dbconn
+    from gppylib.userinput import *
+    from gppylib.system.environment import GpCoordinatorEnvironment
+    from gppylib.system import configurationInterface, configurationImplGpdb
+    from gppylib.programs.clsRecoverSegment_triples import get_segments_with_running_basebackup
+    from gppylib.parseutils import line_reader, check_values, canonicalize_address
+    from gppylib.operations.update_pg_hba_on_segments import update_pg_hba_on_segments
+    from gprebalance_modules.rebalance import GPRebalance
+    from gprebalance_modules.rebalance_validator import ClusterValidator, StateValidationError
+except ImportError as e:
+    sys.exit('ERROR: Cannot import modules.  Please check that you have sourced greenplum_path.sh.  Detail: ' + str(e))
+
+
+DBNAME = 'postgres'
+
+_description = ("""
+Rebalances the existing segments configuration for getting
+optimal perfomance of whole cluster.
+""")
+
+_help = ["""
+The hosts configuration file (YAML format) defines the hosts configuration across the
+cluster. Existing info can be generated from gp_segment_configuration through -g option.
+hosts:
+- hostname: <host_name>      # Machine hostname
+  address: <address>         # Network address for connections
+  primary_datadirs:         # Directories for primary segments
+    - /datadir1
+    - /datadir1
+  mirror_datadirs:          # Directories for mirror segments
+    - /mdatadir1
+    - /mdatadir2
+You can include empty hosts in the configuration, that  is useful
+for planning future cluster balance and segment redistribution.
+         """
+         ]
+
+_usage = """
+ gprebalance -g
+ gprebalance [-m <mode>] [-c] [-f <hosts_file>] [-s] [-v] 
+ gprebalance -c
+ gprebalance -? | -h | --help | --verbose | -v
+"""
+
+EXECNAME = os.path.split(__file__)[-1]
+
+
+def parseargs():
+    parser = OptParser(option_class=OptChecker,
```

Comment:
according to the official docs, [optparse](https://docs.python.org/dev/library/optparse.html) is deprecated in favor of [argparse](https://docs.python.org/dev/library/argparse.html)

Reply:
I agree, but OptParser class is part of gppylib infrastructure, and  I don't think it's crucial for now